fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProvider.java

```diff
@@ -648,11 +674,58 @@ private TopDocs getPageOfData(
             int perPage,
             int currentPage
     ) {
+        /**
+         * Works around the lack of a way to get at the underlying Collector in TimeLimitingCollector
+         */
+        class AccessibleTimeLimitingCollector extends TimeLimitingCollector {
+            private TopScoreDocCollector wrappedCollector;
+
+            /**
+             * @param collector the wrapped collector
+             * @param clock the clock used to implement the timeout
+             * @param ticksAllowed the amount of time allowed for the search
+             */
+            AccessibleTimeLimitingCollector(TopScoreDocCollector collector, Counter clock, long ticksAllowed) {
+                super(collector, clock, ticksAllowed);
+                wrappedCollector = collector;
+            }
+
+            /**
+             * This is the method TimeLimitingCollector was missing.
+             * @return the wrapped collector
+             */
+            public TopScoreDocCollector getWrappedCollector() {
+                return wrappedCollector;
+            }
+        }
+
+        /**
+         * This parallelizes the lucene search - each thread using a time limited collector.
+         */
+        final CollectorManager<AccessibleTimeLimitingCollector, TopDocs> manager
+                = new CollectorManager<AccessibleTimeLimitingCollector, TopDocs>() {
+
+            @Override
+            public AccessibleTimeLimitingCollector newCollector() throws IOException {
+                return new AccessibleTimeLimitingCollector(
+                        TopScoreDocCollector.create(perPage, lastEntry), Counter.newCounter(false), searchTimeout
+                );
+            }
+
+            @Override
+            public TopDocs reduce(Collection<AccessibleTimeLimitingCollector> collectors) throws IOException {
+                final TopDocs[] docs = collectors.stream()
+                        .map((collector) -> collector.getWrappedCollector().topDocs())
+                        .collect(Collectors.toList())
+                        .toArray(new TopDocs[collectors.size()]);
+
+                return TopDocs.merge(perPage, docs);
+            }
+        };
```

Comment:
I think it makes sense to pull out the non-anonymous class since it is named.  But I think the anonymous class makes more sense to leave inside the function.

Pulling out the anonymous class means it has to be named and it also pollutes the list of nouns/classes in the source tree (increasing the mental model of that package).

I think if someone else wants this functionality, it can be refactored at that time.